airflow/providers/google/cloud/transfers/salesforce_to_gcs.py

```diff
@@ -0,0 +1,122 @@
+    :param export_format: Desired format of files to be exported.
+    :type export_format: str
+    :param coerce_to_timestamp: True if you want all datetime fields to be converted into Unix timestamps.
+        False if you want them to be left in the same format as they were in Salesforce.
+        Leaving the value as False will result in datetimes being strings. Default: False
+    :type coerce_to_timestamp: bool
+    :param record_time_added: True if you want to add a Unix timestamp field
+        to the resulting data that marks when the data was fetched from Salesforce. Default: False
+    :type record_time_added: bool
+    :param gzip: Option to compress local file or file data for upload
+    :type gzip: bool
+    :param gcp_conn_id: the name of the connection that has the parameters we need to connect to GCS.
+    :type conn_id: str
+    """
+
+    template_fields = (
+        'query',
+        'bucket_name',
+        'object_name',
+    )
+    template_ext = ('.sql',)
+
+    def __init__(
+        self,
+        *,
+        query: str,
+        bucket_name: str,
+        object_name: str,
+        salesforce_conn_id: str,
+        include_deleted: bool = False,
+        query_params: Optional[dict] = None,
+        export_format: str = "csv",
+        coerce_to_timestamp: bool = False,
+        record_time_added: bool = False,
+        gzip: bool = False,
+        gcp_conn_id: str = "google_cloud_default",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.query = query
+        self.bucket_name = bucket_name
+        self.object_name = object_name
+        self.salesforce_conn_id = salesforce_conn_id
+        self.export_format = export_format
+        self.coerce_to_timestamp = coerce_to_timestamp
+        self.record_time_added = record_time_added
+        self.gzip = gzip
+        self.gcp_conn_id = gcp_conn_id
+        self.include_deleted = include_deleted
+        self.query_params = query_params
+
+    def execute(self, context: Dict):
+        salesforce = SalesforceHook(conn_id=self.salesforce_conn_id)
+        response = salesforce.make_query(
+            query=self.query, include_deleted=self.include_deleted, query_params=self.query_params
+        )
+
+        with tempfile.TemporaryDirectory() as tmp:
+            path = os.path.join(tmp, "salesforce_temp_file")
+            salesforce.write_object_to_file(
+                query_results=response["records"],
+                filename=path,
+                fmt=self.export_format,
+                coerce_to_timestamp=self.coerce_to_timestamp,
+                record_time_added=self.record_time_added,
+            )
+
+            hook = GCSHook(gcp_conn_id=self.gcp_conn_id)
+            hook.upload(
+                bucket_name=self.bucket_name,
+                object_name=self.object_name,
+                filename=path,
+                gzip=self.gzip,
+            )
+            self.log.info("%s uploaded to GCS", path)
```

Comment:
It would be nice to return the path to file so it can be easily referenced in downstream operators

Reply:
hi @turbaszek ,  'path' variable here represents the temporary file that gets created 
just to be uploaded to GCS and wouldn't be available outside of this context.  I'm wondering if it would be more appropriate to pass 
this or alternatively the path to the actual GCS object?  Also wondering about which would be better to log actually.

Reply:
Sorry for not being precise, by returning `path` I thought about returning GCS uri

Reply:
Ok. I implemented uri return value for execute method.